### PR TITLE
Feature/msstats

### PR DESCRIPTION
**Pull request by lukaszimmermann**
PR for `MSstatsConverter`. Based on #3332 . I wil rebase this branch on #3332  once the underlying branch for the experimental design changes.

### CODE REVIEW

**Comment by lukaszimmermann** [comment]
This was already done here #3332 , I'd suggest rebasing on this branch

**Comment by timosachsenberg** [comment]
Renamed Run Section to MSFileSection

**Comment by hroest** [comment]
please have a look at the pyOpenMS as well, the API also needs to change there

**Comment by jpfeuffer** [comment]
`/.../OpenMS/src/utils/MSstatsConverter.cpp:448:61: error: ‘intensity’ may be used uninitialized in this function [-Werror=maybe-uninitialized]`

**Comment by hroest** [comment]
`/.../OpenMS/src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h:121:12: error: ‘std::map’ has not been declared`

**Comment by timosachsenberg** [comment]
Hi Hannes .. working on that. Thanks for taking a look at it

File: src/tests/class_tests/openms/source/ExperimentalDesignFile_test.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF 
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// 
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg$
+// $Authors: Timo Sachsenberg$
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+<<<<<<< HEAD
```
Comment (hroest) [inline src/tests/class_tests/openms/source/ExperimentalDesignFile_test.cpp]: merge conflict?

**Comment by hroest** [comment]
Do you have a test for the tool?

**Comment by jpfeuffer** [comment]
https://abibuilder.informatik.uni-tuebingen.de/jenkins/view/OpenMS/job/openms_user_experimental/job/openms_test_packaging/7/
Sth wrong here

**Comment by jpfeuffer** [comment]
`error message: the value 'MSstats_Condition' was used but is not valid; Registering a required StringOption param (msstats_condition) with a non-empty default is forbidden! [C:\jenkins\ws\openms_user_experimental\openms_test_packaging\c2e226b2\build\doc\doc_param_internal.vcxproj]`

And FileConverter tests still fail: http://cdash.openms.de/viewTest.php?onlyfailed&buildid=266197

File: src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          Single header line
+            Fraction_Group:           Index used to group fractions and source files.
+                                      Note: For label-free this has same cardinality as sample.
+                                      For multiplexed experiments, these might differ as multiple samples can be measured in single files
+            Fraction:                    1st, 2nd, .., fraction. Note: All runs must have the same number of fractions.
+            Spectra_Filepath:         Path to mzML files
+            Label:                    Label in MS file:
+                                        label-free: always 1
+                                        TMT6Plex: 1..6
+                                        SILAC with light and heavy: 1..2
```
[the remark this answers is not in the document]
Comment (timosachsenberg) [reply src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h]: Just an example. In that case you would have 3 labels

**Comment by jpfeuffer** [comment]
Can someone merge in develop please? We had changes in the win installer.

**Comment by timosachsenberg** [comment]
merged develop

**Comment by hroest** [comment]
can you fix this? 

```
src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp
Function parameter 'sample_to_rowindex' should be passed by reference.
275

START_SECTION(([ExperimentalDesign::SampleSection] SampleSection(std::vector< std::vector< String > > _content, std::map< unsigned, Size > sample_to_rowindex, std::map< String, Size > columnname_to_columnindex)))
```

**Comment by timosachsenberg** [comment]
Thanks. I will give this branch another go tomorrow.

**Comment by hroest** [comment]
looks like windows crashed but everything else works

File: src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  TEST_EQUAL(b, true);
+
+  b = fourplex_fractionated_design.sameNrOfMSFilesPerFraction(); 
+  TEST_EQUAL(b, true);
+}
+END_SECTION
+
+START_SECTION((static ExperimentalDesign fromConsensusMap(const ConsensusMap &c)))
+{
+  // TODO
```
Comment (hroest) [inline src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp]: TODO: fix or delete all these tests

File: src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	4	4	1
+	5	1	2
+	6	2	2
+	7	3	2
+	8	4	2
+
+  @ingroup Format
+  */
+
+  class OPENMS_DLLAPI ExperimentalDesignFile
```
Comment (hroest) [inline src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h]: can you add a .pxd file for this as well?

File: src/tests/class_tests/openms/source/ExperimentalDesignFile_test.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+START_SECTION(~ExperimentalDesignFile())
+{
+  delete ptr;
+}
+END_SECTION
+
+START_SECTION((static ExperimentalDesign load(const String &tsv_file, bool require_spectra_files)))
+{
+ExperimentalDesign design = ExperimentalDesignFile::load(
+  OPENMS_GET_TEST_DATA_PATH("ExperimentalDesign_input_1.tsv"), false);
```
Comment (hroest) [inline src/tests/class_tests/openms/source/ExperimentalDesignFile_test.cpp]: can you actually test something here? e.g. what should be in `design`?

File: src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+{
+  bool b = labelfree_unfractionated_design.isFractionated(); 
+  TEST_EQUAL(b, false);
+
+  b = fourplex_fractionated_design.isFractionated(); 
+  TEST_EQUAL(b, true);
+}
+END_SECTION
+
+START_SECTION((bool sameNrOfMSFilesPerFraction() const ))
```
Comment (hroest) [inline src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp]: make sure you test all public methods

File: src/tests/topp/CMakeLists.txt [hunk 7]
```diff
@@ -397,11 +397,11 @@ add_test("TOPP_FileConverter_26_out" ${DIFF} -in1 FileConverter_26.tmp -in2 ${DA
...
-add_test("TOPP_FileConverter_27_out" ${DIFF} -in1 FileConverter_27.tmp -in2 ${DATA_DIR_TOPP}/FileConverter_27_output.mzML -whitelist "location=")
+add_test("TOPP_FileConverter_27_out" ${DIFF} -in1 FileConverter_27.tmp -in2 ${DATA_DIR_TOPP}/FileConverter_27_output.mzML -whitelist "location=" "<offset")
 set_tests_properties("TOPP_FileConverter_27_out" PROPERTIES DEPENDS "TOPP_FileConverter_27")
 # Purpose: test conversion of chromatogram data in minutes to data in seconds
 add_test("TOPP_FileConverter_28" ${TOPP_BIN_PATH}/FileConverter -test -in ${DATA_DIR_TOPP}/FileConverter_28_input.mzML -out FileConverter_28.tmp -out_type mzML  -convert_to_chromatograms)
-add_test("TOPP_FileConverter_28_out" ${DIFF} -in1 FileConverter_28.tmp -in2 ${DATA_DIR_TOPP}/FileConverter_28_output.mzML -whitelist "location=")
+add_test("TOPP_FileConverter_28_out" ${DIFF} -in1 FileConverter_28.tmp -in2 ${DATA_DIR_TOPP}/FileConverter_28_output.mzML -whitelist "location=" "<offset")
 set_tests_properties("TOPP_FileConverter_28_out" PROPERTIES DEPENDS "TOPP_FileConverter_28")
```
Comment (hroest) [inline src/tests/topp/CMakeLists.txt]: can you add tests for the MSStatsConverter?

**Comment by timosachsenberg** [comment]
continued in my branch

File: src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h (hunk 4 above)
Comment (timosachsenberg) [reply src/openms/include/OpenMS/FORMAT/ExperimentalDesignFile.h]: done

File: src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp (hunk 3 above)
Comment (timosachsenberg) [reply src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp]: added test for consensusmap and featuremap

File: src/tests/class_tests/openms/source/ExperimentalDesignFile_test.cpp (hunk 5 above)
Comment (timosachsenberg) [reply src/tests/class_tests/openms/source/ExperimentalDesignFile_test.cpp]: its tested in ExperimentalDesign_test. From my side that's fine but I can also test it here if you want

File: src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp (hunk 6 above)
Comment (timosachsenberg) [reply src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp]: All are tested now, except the simple getter and setter of the inner struct and loading from identifications only data.

File: src/tests/topp/CMakeLists.txt (hunk 7 above)
Comment (timosachsenberg) [reply src/tests/topp/CMakeLists.txt]: will do

**Pull request closed without merging**